src/sync/commits.ts

```diff
@@ -13,15 +17,44 @@ export default async (github: GitHubAPI, repository, cursor, perPage: number) =>
 
 	const refName = (data.repository.defaultBranchRef) ? data.repository.defaultBranchRef.name : "master";
 
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	let commitsData: any = {};
+
 	// TODO: fix typings for graphql
-	const commitsData = (await github.graphql(getCommitsQuery, {
-		owner: repository.owner.login,
-		repo: repository.name,
-		per_page: perPage,
-		cursor,
-		default_ref: refName
-		// eslint-disable-next-line @typescript-eslint/no-explicit-any
-	})) as any;
+	try {
+		commitsData = (await github.graphql(getCommitsQuery(true), {
+			owner: repository.owner.login,
+			repo: repository.name,
+			per_page: perPage,
+			cursor,
+			default_ref: refName
+			// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		})) as any;
+	} catch (err) {
+
+		if (!await booleanFlag(BooleanFlags.RETRY_WITHOUT_CHANGED_FILES, false)) {
+			throw err;
+		}
+
+		// According to the logs, GraphQL queries sometimes fail because the "changedFiles" field is not available.
+		// In this case we just try again, but without asking for the changedFiles field.
+
+		logger.info("retrying without changedFiles");
+
+		const changedFilesErrors = err.errors?.filter(e => e.message?.includes("The changedFiles count for this commit is unavailable"));
+		if (changedFilesErrors.length) {
+			commitsData = (await github.graphql(getCommitsQuery(false), {
```

Comment:
Is there a way to pull this out into a function so we're not repeating the same logic?

Reply:
done